i6_models/parts/frontend/generic_frontend.py

```diff
@@ -0,0 +1,224 @@
+                    in_dim=last_feat_dim,
+                    filter_size=conv_kernel_size if isinstance(conv_kernel_size, int) else conv_kernel_size[1],
+                    stride=conv_stride if isinstance(conv_stride, int) else conv_stride[1],
+                    padding=conv_padding if isinstance(conv_padding, int) else conv_padding[1],
+                )
+                conv_layer_index += 1
+
+            elif layer_type == FrontendLayerType.Pool2d:
+                pool_stride = 1 if model_cfg.pool_strides is None else model_cfg.pool_strides[pool_layer_index]
+                pool_kernel_size = model_cfg.pool_kernel_sizes[pool_layer_index]
+                pool_padding = (
+                    get_same_padding(pool_kernel_size)
+                    if model_cfg.pool_paddings is None
+                    else model_cfg.pool_paddings[pool_layer_index]
+                )
+
+                self.frontend_layers.append(
+                    nn.MaxPool2d(
+                        kernel_size=pool_kernel_size,
+                        stride=pool_stride,
+                        padding=pool_padding,
+                    )
+                )
+                last_feat_dim = calculate_output_dim(
+                    in_dim=last_feat_dim,
+                    filter_size=pool_kernel_size if isinstance(pool_kernel_size, int) else pool_kernel_size[1],
+                    stride=pool_stride if isinstance(pool_stride, int) else pool_stride[1],
+                    padding=pool_padding if isinstance(pool_padding, int) else pool_padding[1],
+                )
+                pool_layer_index += 1
+
+            elif layer_type == FrontendLayerType.Activation:
+                self.frontend_layers.append(model_cfg.activations[activation_layer_index])
+                activation_layer_index += 1
+            else:
+                raise NotImplementedError
+
+            self.linear = nn.Linear(
+                in_features=last_feat_dim * last_channel_dim,
+                out_features=model_cfg.out_features,
+                bias=True,
+            )
+
+    def forward(self, tensor: torch.Tensor, sequence_mask: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        assert tensor.shape[-1] == self.cfg.in_features
+        # and add a dim
+        tensor = tensor[:, None, :, :]  # [B,C=1,T,F]
+
+        for i in range(len(self.cfg.layer_ordering)):
+            layer = self.frontend_layers[i]
+            tensor = layer(tensor)
+
+            if isinstance(layer, nn.Conv2d) or isinstance(layer, nn.MaxPool2d):
+                sequence_mask = mask_pool(
+                    sequence_mask,
+                    kernel_size=layer.kernel_size if isinstance(layer.kernel_size, int) else layer.kernel_size[0],
+                    stride=layer.stride if isinstance(layer.stride, int) else layer.stride[0],
+                    padding=layer.padding if isinstance(layer.padding, int) else layer.padding[0],
```

Comment:
```suggestion
                    kernel_size=layer.kernel_size[0],
                    stride=layer.stride[0],
                    padding=layer.padding[0],
```
If you decide to adopt the "tuple-only-parameter" suggestion.